operator/fee_recipient/controller_test.go

```diff
@@ -52,32 +52,47 @@ func TestSubmitProposal(t *testing.T) {
 	t.Run("submit first time or halfway through epoch", func(t *testing.T) {
 		numberOfRequests := 4
 		var wg sync.WaitGroup
+		wg.Add(numberOfRequests) // Set up the wait group before starting goroutines
+
 		client := beacon.NewMockBeaconNode(ctrl)
 		client.EXPECT().SubmitProposalPreparation(gomock.Any()).DoAndReturn(func(feeRecipients map[phase0.ValidatorIndex]bellatrix.ExecutionAddress) error {
 			wg.Done()
 			return nil
-		}).MinTimes(numberOfRequests).MaxTimes(numberOfRequests) // call first time and on the halfway through epoch. each time should be 2 request as we have two batches
+		}).Times(numberOfRequests)
 
-		ticker := mocks.NewMockTicker(ctrl)
-		ticker.EXPECT().Subscribe(gomock.Any()).DoAndReturn(func(subscription chan phase0.Slot) event.Subscription {
-			subscription <- 1 // first time
-			time.Sleep(time.Millisecond * 500)
-			subscription <- 2 // should not call submit
-			time.Sleep(time.Millisecond * 500)
-			subscription <- 20 // should not call submit
-			time.Sleep(time.Millisecond * 500)
-			subscription <- phase0.Slot(network.SlotsPerEpoch()) / 2 // halfway through epoch
-			time.Sleep(time.Millisecond * 500)
-			subscription <- 63 // should not call submit
-			return nil
-		})
+		ticker := mocks.NewMockSlotTicker(ctrl)
+		mockTimeChan := make(chan time.Time)
+		mockSlotChan := make(chan phase0.Slot)
+		ticker.EXPECT().Next().Return(mockTimeChan).AnyTimes()
+		ticker.EXPECT().Slot().DoAndReturn(func() phase0.Slot {
+			return <-mockSlotChan
+		}).AnyTimes()
 
 		frCtrl.beaconClient = client
-		frCtrl.ticker = ticker
+		frCtrl.slotTickerProvider = func() slot_ticker.SlotTicker {
+			return ticker
+		}
 
 		go frCtrl.Start(logger)
-		wg.Add(numberOfRequests)
+
+		slots := []phase0.Slot{
+			1,                                        // first time
+			2,                                        // should not call submit
+			20,                                       // should not call submit
+			phase0.Slot(network.SlotsPerEpoch()) / 2, // halfway through epoch
+			63,                                       // should not call submit
+		}
+
+		for _, s := range slots {
+			mockTimeChan <- time.Now()
+			mockSlotChan <- s
+			time.Sleep(time.Millisecond * 500)
```

Comment:
Can the duration be reduced? I think it might be too much for the test unless there's any reason for this (let's add a comment in this case)

Reply:
Oh, I didn't notice that. Then I agree it's fine to keep the behavior